src/react/providers/chat-room-provider.tsx

```diff
@@ -77,42 +128,99 @@ export const ChatRoomProvider: React.FC<ChatRoomProviderProps> = ({
 }) => {
   const client = useChatClient();
   const logger = useLogger();
-  logger.trace(`ChatRoomProvider();`, { roomId, options, release, attach });
+  logger.debug(`ChatRoomProvider();`, { roomId, options, release, attach });
+
+  // Set the initial room promise, we do this in a function to avoid rooms.get being called
+  // every time the component re-renders
+  // In StrictMode this will be called twice one after the other, but that's ok
+  const [value, setValue] = useState<ChatRoomContextType>(() => {
+    logger.debug(`ChatRoomProvider(); initializing value`, { roomId, options });
+    return { room: client.rooms.get(roomId, options).catch(), roomId: roomId, options: options, client: client };
+  });
 
-  const [value, setValue] = useState({ room: client.rooms.get(roomId, options) });
+  // Create a queue to manage release ops
+  const roomReleaseQueue = useRef(new RoomReleaseQueue(logger));
 
+  // update the release queue if the logger changes - as it means we have a new client
+  // and only if it actually changes, not because strict mode ran it twice
   useEffect(() => {
-    const room = client.rooms.get(roomId, options);
+    // Don't create a new queue if the logger hasn't actually changed
+    if (roomReleaseQueue.current.logger === logger) {
+      return;
+    }
+
+    logger.debug(`ChatRoomProvider(); updating release queue`);
+    roomReleaseQueue.current = new RoomReleaseQueue(logger);
+  }, [logger]);
+
+  // Create an effect that manages the room state, handles attaching and releasing
+  useEffect(() => {
+    logger.debug(`ChatRoomProvider(); running lifecycle useEffect`, { roomId: roomId });
+    let unmounted = false;
+    let resolvedRoom: Room | undefined;
+    const currentReleaseQueue = roomReleaseQueue.current;
 
-    // Update state if room instance has changed.
-    setValue((prev) => {
-      if (prev.room === room) {
+    // If there was a previous release queued for this room (same id and options), abort it
+    currentReleaseQueue.abort(roomId, options);
+
+    // Get the room promise
+    const room = client.rooms.get(roomId, options).catch();
```

Comment:
Same here regarding the `catch()`.

Reply:
Have changed!

Reply:
I see the empty func catch below but did you forget the `.catch()` on this line (169)?

Reply:
Ooops... fixed!